### PR TITLE
Disable the access to login page for already logged in customers/admins

### PR DESCRIPTION
**Pull request by pjedrzejewski**
| Q               | A
| --------------- | ---
| Bug fix?        | yes |
| New feature?    | yes |
| BC breaks?      | no |
| Related tickets | fixes https://github.com/Sylius/Sylius/issues/3448 |
| License         | MIT |

### CODE REVIEW

File: src/Sylius/Bundle/UiBundle/Controller/SecurityController.php [hunk 1]
```diff
@@ -62,6 +81,13 @@ public function __construct(
      */
     public function loginAction(Request $request): Response
     {
+        $alreadyLoggedInRedirectRoute = $request->attributes->get('_sylius')['logged_in_route'] ?? null;
+
+        if ($alreadyLoggedInRedirectRoute && $this->authorizationChecker->isGranted('IS_AUTHENTICATED_FULLY')) {
+
```
Comment (Zales0123) [inline src/Sylius/Bundle/UiBundle/Controller/SecurityController.php]: :(

Comment (pamil) [reply src/Sylius/Bundle/UiBundle/Controller/SecurityController.php]: :(

Comment (Arminek) [reply src/Sylius/Bundle/UiBundle/Controller/SecurityController.php]: 💃

File: src/Sylius/Behat/Context/Ui/Shop/AccountContext.php [hunk 2]
```diff
@@ -432,6 +441,22 @@ public function iShouldBeAbleToChangePaymentMethodForThisOrder(OrderInterface $o
...
     /**
+     * @Then I should be redirected to my account dashboard
+     */
+    public function iShouldBeRedirectedToMyAccountDashboard()
+    {
+        Assert::true($this->dashboardPage->isOpen(), 'User should be on the account panel dashboard page but they are not.');
+    }
+
+    /**
+     * @When I want to log in
```
Comment (lchrusciel) [inline src/Sylius/Behat/Context/Ui/Shop/AccountContext.php]: I would go with `When I try to log in` and add this step to shop `LoginContext`. Fewer dependencies here and more coherent `LoginContext`.

Comment (pjedrzejewski) [reply src/Sylius/Behat/Context/Ui/Shop/AccountContext.php]: I did that initially but after including the LoginContext in the account suite, I got conflict of step definitions, so I decided to go this way and avoid too many changes. Are we ok with reworking that post release? :)

File: src/Sylius/Behat/Context/Ui/Shop/AccountContext.php [hunk 3]
```diff
@@ -432,6 +441,22 @@ public function iShouldBeAbleToChangePaymentMethodForThisOrder(OrderInterface $o
     }
 
     /**
+     * @Then I should be redirected to my account dashboard
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Ui/Shop/AccountContext.php]: IMO `@Then` steps should be after `@When` steps in context file

**Comment by Mipme** [comment]
Great to hear! Does it also disable access to the register page?

**Pull request merged**